controllers/common/predicate.go

```diff
@@ -0,0 +1,45 @@
+package common
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// ComposePredicates will compose a variable number of predicates and return a predicate that
+// will allow events that are allowed by any of the given predicates.
+func ComposePredicates(predicates ...predicate.Predicate) predicate.Predicate {
```

Comment:
why is this predicate required? we already receive all events for MirrorPeer.

Reply:
I added this to ignore events for status updates. ocs operator also follows this pattern for  storagecluster. Let me know if you want it removed

Reply:
We need to use generation change predicate for that. it comes with controller runtime. Use that.
Maintaining our own version is not needed.

Reply:
Yes, in mirrorpeercontroller i am using the generation change predicate from the predicate package (hope you mean the same). This is the utility function to compose the predicate with the crud funcs.  I am not maintaining anything here.